pkg/platform/aks/aks.go

```diff
@@ -0,0 +1,354 @@
+)
+
+// workerPool defines "worker_pool" block.
+type workerPool struct {
+	// Label field.
+	Name string `hcl:"name,label"`
+
+	// Block properties.
+	Count  int               `hcl:"count,optional"`
+	VMSize string            `hcl:"vm_size,optional"`
+	Labels map[string]string `hcl:"labels,optional"`
+	Taints []string          `hcl:"taints,optional"`
+}
+
+// config defines "cluster" block for AKS.
+type config struct {
+	AssetDir    string            `hcl:"asset_dir,optional"`
+	ClusterName string            `hcl:"cluster_name,optional"`
+	Tags        map[string]string `hcl:"tags,optional"`
+
+	// Azure specific.
+	TenantID       string `hcl:"tenant_id,optional"`
+	SubscriptionID string `hcl:"subscription_id,optional"`
+	ClientID       string `hcl:"client_id,optional"`
+	ClientSecret   string `hcl:"client_secret,optional"`
+
+	Location string `hcl:"location,optional"`
+
+	// ApplicationName for created service principal.
+	ApplicationName string `hcl:"application_name,optional"`
+
+	ResourceGroupName   string `hcl:"resource_group_name,optional"`
+	ManageResourceGroup bool   `hcl:"manage_resource_group,optional"`
+
+	WorkerPools []workerPool `hcl:"worker_pool,block"`
+}
+
+const (
+	name = "aks"
+
+	// Environment variables use to load sensitive parts of the configuration.
+	clientIDEnv       = "LOKOMOTIVE_AKS_CLIENT_ID"
+	clientSecretEnv   = "LOKOMOTIVE_AKS_CLIENT_SECRET" // #nosec G101
+	subscriptionIDEnv = "LOKOMOTIVE_AKS_SUBSCRIPTION_ID"
+	tenantIDEnv       = "LOKOMOTIVE_AKS_TENANT_ID"
+)
+
+// init registers aks as a platform.
+func init() { //nolint:gochecknoinits
+	c := &config{
+		Location:            "West Europe",
+		ManageResourceGroup: true,
+	}
+
+	platform.Register(name, c)
+}
+
+// LoadConfig loads configuration values into the config struct from given HCL configuration.
+func (c *config) LoadConfig(configBody *hcl.Body, evalContext *hcl.EvalContext) hcl.Diagnostics {
+	if configBody == nil {
+		emptyConfig := hcl.EmptyBody()
+		configBody = &emptyConfig
+	}
+
+	if d := gohcl.DecodeBody(*configBody, evalContext, c); d.HasErrors() {
+		return d
+	}
+
+	return c.checkValidConfig()
+}
+
+// checkValidConfig validates cluster configuration.
+func (c *config) checkValidConfig() hcl.Diagnostics {
+	var d hcl.Diagnostics
+
+	// Collect all validation functions here.
+	f := []func() hcl.Diagnostics{
+		c.checkNotEmptyWorkers,
+		c.checkWorkerPoolNamesUnique,
+		c.checkWorkerPools,
+		c.checkCredentials,
+		c.checkRequiredFields,
+	}
+
+	// Then call all of them and collect the diagnostics.
+	for _, vf := range f {
+		d = append(d, vf()...)
+	}
+
+	return d
+}
+
+// checkWorkerPools validates all configured worker pool fields.
+func (c *config) checkWorkerPools() hcl.Diagnostics {
+	var d hcl.Diagnostics
+
+	for _, w := range c.WorkerPools {
+		if w.VMSize == "" {
+			d = append(d, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  fmt.Sprintf("pool %q: VMSize field can't be empty", w.Name),
```

Comment:
Really? There is no default in terraform? Or we don't want to use it?

Reply:
I don't use Terraform module nor variable, so I'd have to do some magic to pick the default size, if pool does not have one defined. Perhaps I could do that.

Reply:
Ups, sorry for the quick comment, then :D